lib/web_ui/dev/integration_tests_manager.dart

```diff
@@ -290,9 +353,40 @@ class IntegrationTestsManager {
     }
   }
 
+  bool get _buildModeSelected =>
+      !IntegrationTestsArgumentParser.instance.buildMode.isEmpty;
+
+  bool get _renderingBackendSelected =>
+      !IntegrationTestsArgumentParser.instance.renderingBackend.isEmpty;
```

Comment:
Same for the rendering backend. `felt` should run in one mode (it could provide `--web-renderer`, just like `flutter run` and `flutter drive` do. This would simplify the `felt` code and user-facing feature set, and make it consistent with the Flutter CLI.

Reply:
As I stated above, I'll leave the code as it is.

As opposed to Flutter CLI this code is mainly used by a few developers frequently changing the web engine code. The following scenario is too common and adding a stress to LUCI resources:
- Developer A completes the design and run the tests with only: `felt test` 
- Tests are passing, developer assumes all is fine and push the change
- PR is created, developer will have to wait for a while for PR to get assigned LUCI resources.
- After waiting 20+ minutes for the result, on LUCI, firefox and ios-safari browsers are failing since the developer didn't run `felt test --browser=firefox` and `felt test --browser=ios-safari`.
Negatives:
- LUCI resources are consumed for a test case which could have been seen on easily on local.
- Developer lost 30 minutes to see the PR is not complete, probably already switched context to a new task.

As principle, it's better to design a tool to show the error as early as possible in the development cycle. Therefore, I'm writing code to cover all 3 debug modes and 2 rendering backends. If a developer want to test their initial idea in one mode they can still pass arguments to test in a specific mode (html+debug for example).

Reply:
I have bad news 😅 I already don't run `felt test`, and mostly rely on unit-tests locally, unless I suspect an integration test may be affected.

Realistically, we will reach a point when it's no longer practical to run all tests locally before sending a PR (just like in flutter/flutter it's impractical to run all shards in `test.dart`). In fact, even today if one is on Linux there's no way to run Safari tests locally. So I think standalone `felt test` is not likely to scale as the project matures.